File: it.unibo.boundaryWalk/app/src/main/java/it/unibo/annotatedExample/ApplISSActor.java
/**
 ApplISSActor.java
 =========================================================================
 An annotated class that exploits annotations (instead of inheritance)
 to introduce built-in behavior in declarative way
 =========================================================================
 */
package it.unibo.annotatedExample;


@ISSActorSpec( actorName = "actor0" )
public class ApplISSActor extends ISSActor{

    //@InitSpec
    public void show(){
        //super.show();
        System.out.println( myname + " ApplISSActor | myname= " + myname);
    }

    public static void main(String args[]) throws Exception {
        new ApplISSActor( ); //.show(); ;
    }

}

File: it.unibo.boundaryWalk/app/src/main/kotlin/it/unibo/annotExample/ApplISSActor.kt
/**
 * ApplISSActor.java
 * =========================================================================
 * An annotated class that exploits annotations (instead of inheritance)
 * to introduce built-in behavior in declarative way
 * =========================================================================
 */
package it.unibo.annotExample

@ISSActorSpec(actorName = "actor0")
class ApplISSActor : ISSActor() {
    @InitSpec
    override fun show() {
        //super.show();
        println("$myname ApplISSActor | myname= $myname")
    }

    companion object {
        @Throws(Exception::class)
        @JvmStatic
        fun main(args: Array<String>) {
            ApplISSActor() //.show(); ;
        }
    }
}